packages/@aws-cdk/aws-ecs/lib/cluster.ts

```diff
@@ -1175,21 +1180,24 @@ export class AsgCapacityProvider extends Construct {
 class MaybeCreateCapacityProviderAssociations implements IAspect {
   private scope: Construct;
   private id: string;
-  private capacityProviders: string[]
-  private resource?: CfnClusterCapacityProviderAssociations
+  private capacityProviders: string[];
+  private resource?: CfnClusterCapacityProviderAssociations;
+  private defaultCapacityProviderStrategy?: CapacityProviderStrategy[];
+
 
-  constructor(scope: Construct, id: string, capacityProviders: string[] ) {
+  constructor(scope: Construct, id: string, capacityProviders: string[], defaultCapacityProviderStrategy?: CapacityProviderStrategy[]) {
     this.scope = scope;
     this.id = id;
     this.capacityProviders = capacityProviders;
+    this.defaultCapacityProviderStrategy = defaultCapacityProviderStrategy;
   }
 
   public visit(node: IConstruct): void {
     if (node instanceof Cluster) {
       if (this.capacityProviders.length > 0 && !this.resource) {
         const resource = new CfnClusterCapacityProviderAssociations(this.scope, this.id, {
           cluster: node.clusterName,
-          defaultCapacityProviderStrategy: [],
+          defaultCapacityProviderStrategy: this.defaultCapacityProviderStrategy || [],
```

Comment:
Let's make sure we have a test which exercises this behavior (ie tests both the default `[]` and when `defaultCapacityProviderStrategy` is specified